src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -819,10 +845,10 @@ def _kind_symbol_from_name(name, symbol_table):
                 kind_symbol.datatype = "integer"
         except KeyError:
             # The SymbolTable does not contain an entry for this kind parameter
-            # so create one.
-            # TODO: Issue #584, the statment below can cause double
-            # declarations.
-            kind_symbol = DataSymbol(name, "integer")
+            # so create one. We specify a DeferredInterface as we don't
+            # currently know how this symbol is brought into scope.
+            kind_symbol = DataSymbol(name, "integer",
```

Comment:
Is the code above except KeyError correct? On this line we set the datatype to "integer" and the interface to "deferred" but above the test is for datatype is "deferred"

Reply:
Hopefully this is clearer now that we're using 'Unresolved' for interfaces and 'Deferred' for types. I've also re-written it slightly.